### PR TITLE
Remove logs after deleting the computer

### PR DESCRIPTION
**Pull request by olivergondza**
I have seen jclouds plugin leaving hundreds log directories behind.

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
I doubt regarding this change. Logs deletion is useful, but there're corner cases. 

From the top of my head:
* Somebody deletes the node in a mistake and restores it via job-config-history or other config backup
* Somebody troubleshoots the issue on the Cloud Slave (which is being killed quickly) and needs logs to analyze the behavior

I would make the behavior configurable (checkbox in the slave deletion dialog, API option for others)

**Comment by KostyaSha** [comment]
+1 for deleting logs
- If you deleted file on PC it also an issue, feel free to implement Trash can for Nodes. (like it exists for jobs?)
- If you have many dynamic slaves - your jenkins will be polluted with all useless logs.
Keeping failed node online or reconfigure kill timeout in retention strategy is more useful (FYI in docker plugin 0.9.0-rc1 it is possible to configure and choose retention strategies). Keeping constantly trash on FS is not a solution for troubleshooting.

File: core/src/main/java/hudson/slaves/SlaveComputer.java [hunk 1]
```diff
@@ -655,6 +655,12 @@ protected void kill() {
         super.kill();
         closeChannel();
         IOUtils.closeQuietly(log);
+
+        try {
+            Util.deleteRecursive(getLogFile().getParentFile());
```
Comment (jglick) [inline core/src/main/java/hudson/slaves/SlaveComputer.java]: To defend against catastrophes from bad future refactorings, I would recommend extracting a protected method for the computer’s log directory, to make it clear that what you are deleting is specific to this computer only.

**Comment by jglick** [comment]
Generally :+1:. I would not want to introduce a new configuration option for this. If there is a real risk of useful logs getting lost, perhaps move the log dir to `${java.io.tmpdir}` and `deleteOnExit`?

**Comment by olivergondza** [comment]
I agree there is a potential for loosing important information though the implementation is simpler than *delete all the logs of deleted slaves when too old* approach.

I run into this problem trying to make Beaver to publish my logs to Logstash but monitoring hundreds of logs 99% of which are stale was just too cumbersome.

**Comment by olivergondza** [comment]
Jenkins never guaranteed the useful slave logs will not be rotated before admin can investigate. If someone really cares, it should be possible to capture and save the log from `ComputerListener#onLaunchFailure`.

@oleg-nenashev, is that OK with you?

**Comment by oleg-nenashev** [comment]
@olivergondza 
Yes, I agree with the votes from other reviewers. The functionality can be improved later if there're requests

**Pull request merged**